### ISSUE TITLE
Discussions / Good first issues / Default Components

### ISSUE THREAD

**Issue by TimWhiting** [simolus3/zap#4]
A few friendly suggestions:
- It would be great to enable Discussions on this repo so that questions could be asked that aren't really issues with the framework. 
- Additionally it would be helpful to know what issues you feel someone could work on to get started contributing. I'd be happy to contribute or tackle one of the current issues, but I have no clue if you are planning or in the middle of working on them right now. If you want to work on them or feel like it is a major change that you want to oversee, maybe assign yourself as the Assignee on the issue so newcomers know what they can contribute on.
- Finally I'm wondering if it would make sense to contribute a default set of components in a separate package such as `material_zap` for material components, and if that could be an *official* package hosted in this repo rather than some 3rd party solution. This way the community could come together to contribute high quality components. It would reduce the barrier to entry for contributing, and would help the community grow faster, as well as reduce fragmentation in the ecosystem I think. The default set of material components in flutter for example I think is what makes it so popular and easy to get started working with or contributing to.

 Anyways, glad to see this project!

**Comment by simolus3** [comment]
Thanks for your feedback!

> It would be great to enable Discussions on this repo so that questions could be asked that aren't really issues with the framework.

Good point, thanks! I've enabled discussions.

> Additionally it would be helpful to know what issues you feel someone could work on to get started contributing.

Also a really good point. The current issues are mostly notes to myself and not necessarily great starting points for working on this project... They mostly resolve around the generator which is still a bit hacky at the moment. But of course some incremental issues to refactor the generator could be good first issues, I'll create them as I see fit.
I have also opened two issues around the docs / more general ideas that are a better starting point and would really help the project.

> Finally I'm wondering if it would make sense to contribute a default set of components in a separate package such as `material_zap` for material components, and if that could be an _official_ package hosted in this repo rather than some 3rd party solution.

Yes, I think this would be great! Are you interested in helping out here? I can scaffold a `material_zap` or a `zap_components` package, and I agree that this is a good point to start contributing :)

**Comment by TimWhiting** [comment]
Yeah, I'd be happy to start contributing to a `material_zap` or `zap_components` package. As far as design for these components, I'd like to provide some styling options as a theme like in flutter (possibly exposed via a riverpod provider), in addition for allowing customization. I'm not super familiar however with sass, so I'm wondering if you have suggestions on how to allow for properties / themes for styling, but also allowing for custom css.

**Comment by simolus3** [comment]
> As far as design for these components, I'd like to provide some styling options as a theme like in flutter (possibly exposed via a riverpod provider)

For styling options, the easy solutions I know about are [Sass variables](https://sass-lang.com/documentation/variables) (evaluated at compile-time) or [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties).
I'm slightly leaning towards custom CSS properties because they can be assigned with CSS selectors (so we could assign a different set of variables for different components in an app, or have different values for night mode etc.).

Entirely custom CSS will be a bit harder to do I imagine. We could pass it down as a `@prop` and set it on the relevant elements with an inline `style` attribute, or maybe let users write global CSS that matches some component elements to add the additional styles.
